java/arcs/core/storage/CapabilitiesResolver.kt

```diff
@@ -13,25 +13,58 @@ package arcs.core.storage
 
 import arcs.core.common.ArcId
 import arcs.core.data.Capabilities
+import arcs.core.data.Schema
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
 import arcs.core.util.TaggedLog
 
 /**
  * [CapabilitiesResolver] is a factory class that creates [StorageKey]s based on the registered
  * providers and given [Capabilities].
  */
 class CapabilitiesResolver(
-    val options: StorageKeyOptions,
+    val options: CapabilitiesResolverOptions,
     private val creators: StorageKeyCreatorsMap = getAllCreators()
 ) {
     private val log = TaggedLog { "CapabilitiesResolver" }
 
     /* Options used to construct [CapabilitiesResolver] */
-    data class StorageKeyOptions(val arcId: ArcId)
+    data class CapabilitiesResolverOptions(val arcId: ArcId)
+
+    /* Options passed to registered [StorageKey] constructors */
```

Comment:
Since these are public facing, KDoc'ing all three data classes would be good, and putting @property notifications on the interface fields describing what they do would be good. For example, while arcId and entitySchema are self explanatory, I don't really know off the top of my head what unique and location are being used for.

Reply:
done